index.js

```diff
@@ -211,13 +213,21 @@ Sharp.prototype.flatten = function(flatten) {
 };
 
 Sharp.prototype.overlayWith = function(overlayPath) {
-  if (typeof overlayPath !== 'string') {
-    throw new Error('The overlay path must be a string');
-  }
-  if (overlayPath === '') {
-    throw new Error('The overlay path cannot be empty');
+  try {
+    var colour = color(overlayPath);
+    this.options.overlayColor = colour.rgbArray();
+    this.options.overlayColor.push(colour.alpha() * 255);
```

Comment:
We might want to Math.round the alpha value as it gets clipped to an integer in C++ land. (I've spotted that the existing `background` method should do this also.)